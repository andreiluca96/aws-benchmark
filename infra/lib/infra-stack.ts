import { Duration, Stack, StackProps } from 'aws-cdk-lib';
import * as sns from 'aws-cdk-lib/aws-sns';
import * as subs from 'aws-cdk-lib/aws-sns-subscriptions';
import * as sqs from 'aws-cdk-lib/aws-sqs';
import { Construct } from 'constructs';
import { BenchmarkingResources } from './constructs/benchmarking-resources'
import { Backend } from './constructs/backend'
import { FrontendConstruct } from './constructs/frontend';

export class InfraStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

		const region = 'eu-central-1';

		new BenchmarkingResources(this, 'BenchmarkingResources', {
			 prefix: region
	  })

		new Backend(this, 'BackendResources', {
			prefix: region 
		})

		new FrontendConstruct(this, 'FrontendConstruct', {
			prefix: region
		});
  }
}
