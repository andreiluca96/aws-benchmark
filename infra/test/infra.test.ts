import * as cdk from 'aws-cdk-lib';
import { Template, Match } from 'aws-cdk-lib/assertions';
import * as Infra from '../lib/infra-stack';

test('matches the snapshot', () => {
  
	const app = new cdk.App();
  
	// WHEN
  const stack = new Infra.InfraStack(app, 'MyTestStack');
  
	// THEN
  const template = Template.fromStack(stack);

  expect(template.toJSON()).toMatchSnapshot();
});
