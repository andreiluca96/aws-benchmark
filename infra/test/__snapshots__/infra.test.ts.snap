// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
{
  "Outputs": {
    "BackendResourcesTheUrlDE242F50": {
      "Value": {
        "Fn::GetAtt": [
          "BackendResourcesBenchmarkRunnerFunctionUrlEEC75B51",
          "FunctionUrl",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "BackendResourcesBenchmarkRunner47B6367D": {
      "DependsOn": [
        "BackendResourcesBenchmarkRunnerServiceRoleD04A7008",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "04fffe46efd08c87da3b54d6774a6114b25902c2b4e382781831ff9c6117b1ea.zip",
        },
        "FunctionName": "eu-central-1-benchmark-runner-lambda",
        "Handler": "bootstrap",
        "Role": {
          "Fn::GetAtt": [
            "BackendResourcesBenchmarkRunnerServiceRoleD04A7008",
            "Arn",
          ],
        },
        "Runtime": "provided.al2023",
      },
      "Type": "AWS::Lambda::Function",
    },
    "BackendResourcesBenchmarkRunnerFunctionUrlEEC75B51": {
      "Properties": {
        "AuthType": "NONE",
        "TargetFunctionArn": {
          "Fn::GetAtt": [
            "BackendResourcesBenchmarkRunner47B6367D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Url",
    },
    "BackendResourcesBenchmarkRunnerServiceRoleD04A7008": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "BackendResourcesBenchmarkRunnerinvokefunctionurlA121688A": {
      "Properties": {
        "Action": "lambda:InvokeFunctionUrl",
        "FunctionName": {
          "Fn::GetAtt": [
            "BackendResourcesBenchmarkRunner47B6367D",
            "Arn",
          ],
        },
        "FunctionUrlAuthType": "NONE",
        "Principal": "*",
      },
      "Type": "AWS::Lambda::Permission",
    },
    "BenchmarkingResourcesBenchMarkEventBus062270BA": {
      "Properties": {
        "Name": "eu-central-1-benchmark-event-bus",
      },
      "Type": "AWS::Events::EventBus",
    },
    "BenchmarkingResourcesBenchmarkQueue6E3D0BE7": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "eu-central-1-benchmark-queue",
        "VisibilityTimeout": 300,
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "BenchmarkingResourcesBenchmarkTopic10182B2C": {
      "Properties": {
        "TopicName": "eu-central-1-benchmark-topic",
      },
      "Type": "AWS::SNS::Topic",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
